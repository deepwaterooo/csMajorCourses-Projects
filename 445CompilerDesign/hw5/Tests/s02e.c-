// C-F13
// testing static variables

// toad() is in global, int toad is in local
  toad() { int toad; }  

int load;
// load() is in global, int load is in local but statically allocated
  load() 
{ 
    static int load; 
    int load;
}

bool runningSum;

int sum(bool init; int x) {
    static int runningSum;

    if (init) runningSum = 0;
    else runningSum += x;

    return runningSum;
}

  main()
{
    sum(true, 666);
    sum(false, 3);
    output(sum(false, 0));
}
