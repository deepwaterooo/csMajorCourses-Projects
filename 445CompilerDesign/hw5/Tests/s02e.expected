ERROR(9): Symbol 'load' is already defined at line 7.
ERROR(12): Symbol 'load' is already defined at line 11.
Number of warnings: 0
Number of errors: 2
------------------------------------
